arraycontext/container/traversal.py

```diff
@@ -987,4 +989,79 @@ def treat_as_scalar(x: Any) -> bool:
 
 # }}}
 
+
+# {{{
```

Comment:
Should these be here or in `arithmetic`?

Reply:
I imagine they'll be pretty exclusively imported from the top level.. so, meh? `arithmetic` is probably fine just containing the dunder methods.